bot/seasons/evergreen/reddit.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import random
+
+import discord
+from discord.ext import commands
+from discord.ext.commands.cooldowns import BucketType
+
+from bot.pagination import ImagePaginator
+
+
+log = logging.getLogger(__name__)
+
+
+class Reddit(commands.Cog):
+    """Fetches reddit posts."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    async def fetch(self, url: str) -> dict:
+        """Send a get request to the reddit API and get json response."""
+        session = self.bot.http_session
+        params = {
+            'limit': 50
+        }
+        headers = {
+            'User-Agent': 'Iceman'
+        }
+
+        async with session.get(url=url, params=params, headers=headers) as response:
+            return await response.json()
+
+    @commands.command(name='reddit')
+    @commands.cooldown(1, 10, BucketType.user)
+    async def get_reddit(self, ctx: commands.Context, subreddit: str = 'python', sort: str = "hot") -> None:
+        """
+        Fetch reddit posts by using this command.
+
+        Gets a post from r/python by default.
+        Usage:
+        --> .reddit [subreddit_name] [hot/top/new]
+        """
+        pages = []
+        sort_list = ["hot", "new", "top", "rising"]
+        if sort.lower() not in sort_list:
+            await ctx.send(f"Invalid sorting: {sort}\nUsing default sorting: `Hot`")
+            sort = "hot"
+
+        data = await self.fetch(f'https://www.reddit.com/r/{subreddit}/{sort}/.json')
+
+        try:
+            posts = data["data"]["children"]
+        except KeyError:
+            return await ctx.send('Subreddit not found!')
+        if not posts:
+            return await ctx.send('No posts available!')
+
+        if posts[1]["data"]["over_18"] is True:
+            return await ctx.send("You cannot access this Subreddit.")
+
+        upvote_emoji = self.bot.get_emoji(565557799040319508)
+        comment_emoji = self.bot.get_emoji(565576076483624960)
+        user_emoji = "🎅"
+
+        # embed_titles = discord.Embed(colour=0xf9f586)
+        embed_titles = ""
+
+        random_posts = []
+        while True:
+            if len(random_posts) == 5:
+                break
+            rand_post = random.choice(posts)
+            if rand_post not in random_posts:
+                random_posts.append(rand_post)
```

Comment:
You can shorten this via `random.sample`, for example like this
```py
random_posts = random.sample(posts, k=5)
```
And it will be guaranteed to be 5 unique posts as long as `posts` does not contain duplications.
From its docstring: `Chooses k unique random elements from a population sequence or set.`

Reply:
that's really cool, thanks!